app/models/item.rb

```diff
@@ -1,3 +1,9 @@
 class Item < ApplicationRecord
   belongs_to :merchant
+  has_many :invoice_items
+  has_many :invoices, through: :invoice_items
+  
+  def self.invoice_finder(merchant_id)
+    Invoice.joins(:invoice_items => :item).where(:items => {:merchant_id => merchant_id})
```

Comment:
would this work instead? `Merchant.find(merchant_id).invoices`

Reply:
I don't think it will, Merchants don't have a relationship to invoices, so you can't call `invoices` on a `merchant`